Source/WebCore/platform/ScrollableArea.cpp

```diff
@@ -545,7 +545,7 @@ void ScrollableArea::resnapAfterLayout()
     LOG_WITH_STREAM(ScrollSnap, stream << *this << " resnapAfterLayout: isScrollSnapInProgress " << isScrollSnapInProgress() << " isUserScrollInProgress " << isUserScrollInProgress());
 
     auto* scrollAnimator = existingScrollAnimator();
-    if (!scrollAnimator || isScrollSnapInProgress() || isUserScrollInProgress())
+    if (!scrollAnimator || isScrollSnapInProgress() || isUserScrollInProgress() || !isInStableState())
```

Comment:
This logic makes me feel like `isInStableState()` should encapsulate the in-progress nature of ScrollSnap or UserScroll. I.e., `isInStableState` should probably be false any time a snap or user scroll is in progress.